src/cogent3/core/new_alignment.py

```diff
@@ -356,35 +395,72 @@ def take_seqs(
         negate
             select all sequences EXCEPT names
         kwargs
-            keyword arguments to be passed to the constructor of the new seqcollection
+            keyword arguments to be passed to the constructor of the new collection
+
+        Notes
+        -----
+        The seqs in the new collection will be references to the same objects as
+        the seqs in the old collection.
         """
+        # todo: kath, note gavins comment that this method could operate on only the names
+        # list and not the seqs_data object
 
-        # todo: kath, add subset method onto SeqsData
         if negate:
-            selected_seqs = {
-                name: self.seqs.get_seq_str(seqid=name)
-                for name in self.names
-                if name not in names
-            }
+            names = [name for name in self.names if name not in names]
         else:
-            selected_seqs = {
-                name: self.seqs.get_seq_str(seqid=name)
-                for name in names
-                if name in self.names
-            }
+            names = [name for name in names if name in self.names]
 
-        if not selected_seqs:
+        if not names:
             return {}
 
-        moltype = kwargs.pop("moltype", self.moltype)
+        seqs_data = self.seqs.subset(names)
 
-        seqs_data = SeqsData(data=selected_seqs, alphabet=moltype.alphabet)
+        moltype = kwargs.pop("moltype", self.moltype)
```

Comment:
Delete this. Don't support indirect ways of changing moltype. A `to_moltype()` method should be the only way to transform a moltype of a collection. In doing so, we need to add an abstract method of `to_alphabet()` on `SeqDataABC` and implement that on `SeqData`. This can be easy if the alphabet's are nearly identical, like DNA and RNA, but slightly more involved if they are not. An example of the latter case will be someone has created a SeqData instance without defining the correct moltype. If we default to a general one (like "text") then it's a superset of the others. So we need to convert the stored data into the new index coordinates. There are classes in `new_alphabet.py` to assist with this.